core/src/main/java/org/apache/iceberg/rest/RESTViewOperations.java

```diff
@@ -59,6 +60,8 @@ public void commit(ViewMetadata base, ViewMetadata metadata) {
     // this is only used for replacing view metadata
     Preconditions.checkState(base != null, "Invalid base metadata: null");
 
+    ViewUtil.verifyReplaceStrictness(base, metadata);
```

Comment:
Could this check be done in `ViewMetadata.Builder` instead? That seems like a safer place because it is used by all view catalog implementations.